### DISCUSSION TITLE
Image carousel: allow fetching images dynamically from API

### DISCUSSION THREAD

**Discussion by itay-sho** [elementor/elementor#28224] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

We have an image carousel that we want to be able to change dynamically.
In our case we have an Elementor site and we sells many different products, we have a carousel too that shows some of the product images.
When the products are updated we want to update the images on the carousel without having to change it statically.

### Describe the solution you'd like.

in our case the data for the products is provided by an API, and it could be very easy to add another endpoint that brings list of product images.
We are looking for a feature that gets an API endpoint and build the carousel dynamically
We have an API endpoint that gives list of images urls, and the carousel collects the data from there and updates it.

As far as I understand, this feature is very similar to an existing feature in elementor tables, when you can select multiple sources to fill dynamically the table.

### Describe alternatives you've considered.

Continuing on doing it statically and update it every once in a while.
Another alternative is to have something external to Elementor based on HTML that allows dynamic content.

### Additional context

Right now I don't have more. feel free to ask questions in case you need deeper understanding.
Thank you!!!

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by tiago-sartor** [comment]
You have to use the Loop Carousel (https://elementor.com/help/loop-carousel/) widget.

**Reply by itay-sho** [reply]
Hello Tiago. Thank you for taking a look.
Although really tried and read the documentation I couldn't figure out how to get custom data which is not a wordpress post / page etc.

My data is available from an API endpoint, and couldn't figure a way to allow it. I tried to use [customer query filter](https://developers.elementor.com/docs/hooks/custom-query-filter/) and add some PHP code, but I couldn't figure a way to return my own data (it is always trying to access to the DB).

I would be grateful if you can elaborate more about how you think it is possible to allow my use case using loop carousel. thank you!

**Comment by tiago-sartor** [comment]
I think the easiest way of accomplishing what you want is by using the ACF plugin. This way you can add custom image fields to your products or posts and populate these fields with your images. Then you can pass the images from the ACF fields to the Loop Carousel widget dynamically.

See: https://www.advancedcustomfields.com/resources/wp-rest-api-integration/
